### PR TITLE
solved two issues:

### PR DESCRIPTION
**Pull request by tarzzz**
1. modified server to open new tab in browser in IPython
2. Handle SIGPIPE error in server

### CODE REVIEW

File: pydy_viz/scene.py [hunk 1]
```diff
@@ -348,11 +348,12 @@ def _display_from_ipython(self):
...
         jobs.new('server.run()')
+
         print '''
         Your visualization is being rendered at
-                 http://localhost:%s/
-                 Visit the url in your webgl compatible browser
-                 to see the animation in full glory'''%(server.port)
+        http://localhost:%s/
+        Opening the visualization in new tab...'''%(server.port)
+        webbrowser.open("http://localhost:%s/"%server.port)
```
Comment (moorepants) [inline pydy_viz/scene.py:356]: Does this work whether you are in an IPython notebook or a normal session?

Comment (tarzzz) [inline pydy_viz/scene.py:356]: It works for ipython notebook only.
Otherwise it simply prompts user to go to url manually

Comment (moorepants) [inline pydy_viz/scene.py:356]: ok, that seems fine. i haven't tested this branch yet, will try to this evening. but you can merge if you feel confident it all works correctly cross browser.

**Comment by moorepants** [comment]
I just tried this branch in firefox and a new tab did not open for me.

**Comment by moorepants** [comment]
tried in chromium too and no new tab opened.

**Comment by tarzzz** [comment]
Is it giving you a traceback?
Or blocking ipython like earlier?

**Comment by tarzzz** [comment]
Can you give me output running this command from your IPython notebook?

```
get_ipython().config
```

I think the IPython display method _display_from_ipython() is not being
called in your case.
The problem might be in scene.display()'s method handling.

**Comment by moorepants** [comment]
No traceback, it just blocks like before.

Here is the output:

selection_035

**Comment by tarzzz** [comment]
@moorepants  Ok, I have figured out the problem here.
This is my output of the same command:

<code>{'IPKernelApp': {'connection_file': u'/home/tarun/.config/ipython/profile_default/security/kernel-168c4c48-f9b2-4772-9a66-d4185925f6fe.json',
  'parent': 1},
 'KernelApp': {'parent_appname': 'ipython-notebook'},
 'ProfileDir': {},
 'Session': {'key': u'c2dd21b1-b6e3-4dd8-9d08-f08b6325596c'}} </code>

See the 'parent_appname' key which is arranged differently in your config dict and mine.
I suppose these are getting changed in IPython versions. How do you suggest we handle this problem?
(Refer line 375 in pydy_viz/scene.py)

**Comment by moorepants** [comment]
Ask on the IPython mailing list.

Jason
moorepants.info
+01 530-601-9791

**Comment by tarzzz** [comment]
Now it detects any IPython frontend (which is recommended as mentioned on this Post on StackOverflowhttp://stackoverflow.com/questions/15411967/how-can-i-check-if-code-is-executed-in-the-ipython-notebook)
This should now work on any frontend attached to IPython kernel.

**Comment by moorepants** [comment]
Works for me now! Great.

Merge master to this branch and run a final test. If all is good, then merge to master.

**Comment by moorepants** [comment]
I got this error when I tried to close the server via the gui button:

```
(pydy-dev)moorepants@moorepants-UL30A:notebooks(master)$ ipython notebook
2014-02-20 12:33:19.006 [NotebookApp] Using existing profile dir: u'/home/moorepants/.config/ipython/profile_default'
2014-02-20 12:33:19.088 [NotebookApp] Using MathJax from CDN: http://cdn.mathjax.org/mathjax/latest/MathJax.js
2014-02-20 12:33:19.134 [NotebookApp] Serving notebooks from local directory: /home/moorepants/Conferences/PYCON2014/pydy-tutorial-pycon2014/notebooks
2014-02-20 12:33:19.135 [NotebookApp] The IPython Notebook is running at: http://127.0.0.1:8888/
2014-02-20 12:33:19.135 [NotebookApp] Use Control-C to stop this server and shut down all kernels (twice to skip confirmation).

(process:6602): GLib-CRITICAL **: g_slice_set_config: assertion 'sys_page_size == 0' failed
2014-02-20 12:33:33.765 [NotebookApp] Connecting to: tcp://127.0.0.1:58166
2014-02-20 12:33:33.821 [NotebookApp] Kernel started: 9122bdeb-0168-47f8-bf57-423c886518dc
2014-02-20 12:33:35.771 [NotebookApp] Connecting to: tcp://127.0.0.1:35007
2014-02-20 12:33:35.798 [NotebookApp] Connecting to: tcp://127.0.0.1:35847
2014-02-20 12:33:35.804 [NotebookApp] Connecting to: tcp://127.0.0.1:59861

(process:6727): GLib-CRITICAL **: g_slice_set_config: assertion 'sys_page_size == 0' failed
Unhandled exception in thread started by <bound method ParentPollerUnix.__bootstrap of <ParentPollerUnix(Thread-1, stopped daemon 140092639278848)>>
Error in sys.excepthook:
Traceback (most recent call last):
  File "/home/moorepants/envs/pydy-dev/local/lib/python2.7/site-packages/IPython/core/ultratb.py", line 1030, in __call__
    self.handler((etype, evalue, etb))
  File "/home/moorepants/envs/pydy-dev/local/lib/python2.7/site-packages/IPython/core/ultratb.py", line 1019, in handler
    ostream.write(self.text(etype, evalue, etb))
  File "/home/moorepants/envs/pydy-dev/local/lib/python2.7/site-packages/IPython/core/ultratb.py", line 407, in text
    tb_offset, context)
  File "/home/moorepants/envs/pydy-dev/local/lib/python2.7/site-packages/IPython/core/ultratb.py", line 1080, in structured_traceback
    return VerboseTB.structured_traceback(
AttributeError: 'NoneType' object has no attribute 'structured_traceback'

Original exception was:
Traceback (most recent call last):
  File "/usr/lib/python2.7/threading.py", line 781, in __bootstrap
2014-02-20 12:36:20.325 [NotebookApp] Kernel shutdown: 9122bdeb-0168-47f8-bf57-423c886518dc
```

**Comment by moorepants** [comment]
The error didn't cause anything to fail but it seems that I never get that little pop-up window that says "ok you can close this tab now"

**Comment by tarzzz** [comment]
server is run as a Background job..
When we kill it from the gui button, the thread also gets killed.

**Comment by moorepants** [comment]
Ok, this worked on Windows XP and Firefox for me. Going to merge.

**Pull request merged**